### DISCUSSION TITLE
Using $persist - Storing and retrieving data

### DISCUSSION THREAD

**Discussion by MartijnHarmenzon** [alpinejs/alpine#3206] [1. Help]
I am trying to use $persist to use the data stored with $persist on multiple pages.
I would like to store something on page /page1/index.html and use these values on /page2/index.html.
How can I achieve this. The storing part on page 1 is going fine. But how to retrieve the data on page 2?

**Comment by ekwoka** [comment]
It should be automatic with $persist if the key is the same.

**Comment by MartijnHarmenzon** [comment]
Thanks @ekwoka. I was able to create a working example. Below the code. 

Code page 1
`<div x-data="page1Data()" x-text="user.user_id"></div>`

```
document.addEventListener("alpine:init", () => {
  Alpine.data('page1Data', function () {
    return {
      user: this.$persist({
        user_id: null
      }),

      init() {
        this.user.user_id = 'ABC123';
      }
    }
  })
})
```

Code page 2
`<div x-data="page2Data()" x-text="user.user_id"></div>`

```
document.addEventListener("alpine:init", () => {
  Alpine.data('page2Data', function () {
    return {
      user: this.$persist({
        user_id: null
      })
    }
  })
})
```

**Comment by MartijnHarmenzon** [comment]
Trying it with $store however doesn't work. With the following code, the object is set within localstorage, but the user_id remains null after trying to change it in init().

`<div x-data x-text="$store.userStore.user.user_id"></div>`

```
document.addEventListener("alpine:init", () => {
  Alpine.store('userStore', {
    user: Alpine.$persist({
      user_id: null
    }),

    init() {
      this.user.user_id = 'ABC123';
    }
  })
})
```

**Comment by MartijnHarmenzon** [comment]
Also managed to solve that by defining the object in $store and setting it in $data. Using the folowing code:
```
<div x-data x-text="$store.userStore.user.user_id"></div>
<div x-data="page3Data()"></div>
```

```
document.addEventListener("alpine:init", () => {
  Alpine.store('userStore', {
    user: Alpine.$persist({
      user_id: null
    })
  })
})
```

```
document.addEventListener("alpine:init", () => {
  Alpine.data('page3Data', function () {
    return {
      init() {
        this.$store.userStore.user.user_id = 'ABC123';
      }
    }
  })
})
```
